internal/services/appconfiguration/app_configuration_resource.go

```diff
@@ -434,6 +488,61 @@ func resourceAppConfigurationUpdate(d *pluginsdk.ResourceData, meta interface{})
 		return fmt.Errorf("updating %s: %+v", *id, err)
 	}
 
+	if d.HasChange("replica") {
+		replicaClient := meta.(*clients.Client).AppConfiguration.ReplicasClient
+		operationsClient := meta.(*clients.Client).AppConfiguration.OperationsClient
+
+		// check if a replica has been removed from config and if so, delete it
+		deleteReplicaIds := make([]replicas.ReplicaId, 0)
+		oldReplicas, newReplicas := d.GetChange("replica")
+		for _, oldReplica := range oldReplicas.(*pluginsdk.Set).List() {
+			isRemoved := true
+			oldReplicaMap := oldReplica.(map[string]interface{})
+
+			for _, newReplica := range newReplicas.(*pluginsdk.Set).List() {
+				newReplicaMap := newReplica.(map[string]interface{})
+
+				if strings.EqualFold(oldReplicaMap["name"].(string), newReplicaMap["name"].(string)) && strings.EqualFold(oldReplicaMap["location"].(string), newReplicaMap["location"].(string)) {
+					isRemoved = false
+					break
+				}
+			}
+
+			if isRemoved {
+				deleteReplicaIds = append(deleteReplicaIds, replicas.NewReplicaID(id.SubscriptionId, id.ResourceGroupName, id.ConfigurationStoreName, oldReplicaMap["name"].(string)))
+			}
+		}
+
+		if err := deleteReplicas(ctx, replicaClient, operationsClient, deleteReplicaIds); err != nil {
+			return err
+		}
+
+		expandedReplicas, err := expandAppConfigurationReplicas(d.Get("replica").(*schema.Set).List(), id.ConfigurationStoreName, location.Normalize(existing.Model.Location))
+		if err != nil {
+			return fmt.Errorf("expanding `replica`: %+v", err)
+		}
+
+		// check if a replica has been added or an existing one changed its location, (re)create it
+		for _, replica := range expandedReplicas {
+			replicaId := replicas.NewReplicaID(id.SubscriptionId, id.ResourceGroupName, id.ConfigurationStoreName, *replica.Name)
+
+			existingReplica, err := replicaClient.Get(ctx, replicaId)
+			if err != nil {
+				if !response.WasNotFound(existingReplica.HttpResponse) {
+					return fmt.Errorf("retrieving %s: %+v", replicaId, err)
+				}
+			}
+
+			if !response.WasNotFound(existingReplica.HttpResponse) {
+				continue
```

Comment:
We should be surfacing this as an error to the user because it means that the replica they're trying to add already exists.

Reply:
fixed, thanks.